torch/csrc/jit/script/compiler.cpp

```diff
@@ -687,20 +721,27 @@ Value* emitBuiltinCall(
   std::stringstream failure_messages;
   //first we try to match the schema without any conversion
   //if no schema matches then insert ImplicitTensorToNum
-  for(bool convert_tensors_to_nums : {false, true}) {
-    //clear previous error messages
+  for (bool convert_tensors_to_nums : {false, true}) {
+    // clear previous error messages
     failure_messages.str("");
     for (const std::shared_ptr<Operator>& op : variants) {
-      if (auto result = tryEmitBuiltin(
-              op, failure_messages, loc, graph, name, inputs, attributes,
-              convert_tensors_to_nums)) {
-        return result;
+      const auto matched_inputs = matchStrategy(
```

Comment:
This is huge code smell that we got something wrong here. We now have `Impl` variants of builtin and strategy patterns. All to just get over the fact that there are now two ways of matching schema! There should only be a single way to emit ops and check their schema is valid.

Here is what I think we should do: just make emitBuiltinCall universally insert the world token if the schema is mutable. It should emit a `prim::LoadWorld` op.  `prim::LoadWorld` should just fault on use. It will then be a passes responsibility if they are emitting a mutable op to fix up the prim::LoadWorld nodes. This kicks the can down the road but I think it is better than complicated the `emit` logic in this way. We can improve this later. For instance, we can add a `std::function<Value*()> world_provider = nullptr` to emit that allows you to specify how to generate a World input if needed.